inbc-program/tests/unit/test_source_app_parser.py

```diff
@@ -42,16 +112,38 @@ def test_create_add_manifest_successfully(self, m_connect):
                    '</repos><filename>intel-gpu-jammy.list</filename></repo></add></applicationSource></manifest>'
         self.assertEqual(p.func(p), expected)
 
-    def test_parse_remove_arguments_successfully(self):
+    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
+    def test_create_add_minus_gpg_manifest_successfully(self, m_connect):
+        p = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--sources', 'deb http://example.com/ focal main restricted universe',
+             'deb-src http://example.com/ focal-security main',
+             '--filename', 'intel-gpu-jammy.list'])
+        Inbc(p, 'source', False)
+        expected = '<?xml version="1.0" encoding="utf-8"?><manifest><type>source</type><applicationSource>' \
+                   '<add><repo><repos>' \
+                   '<source_pkg>deb http://example.com/ focal main restricted universe</source_pkg>' \
+                   '<source_pkg>deb-src http://example.com/ focal-security main</source_pkg>' \
+                   '</repos><filename>intel-gpu-jammy.list</filename></repo></add></applicationSource></manifest>'
```

Comment:
**Suggestion:** Avoid using hardcoded XML strings in tests. Use an XML library to create and compare XML documents, which will make the tests less brittle to changes in the XML structure. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        from xml.etree.ElementTree import Element, SubElement, tostring

        def construct_manifest_xml(sources, filename):
            manifest = Element('manifest')
            manifest.set('version', '1.0')
            manifest.set('encoding', 'utf-8')
            application_source = SubElement(manifest, 'applicationSource')
            add = SubElement(application_source, 'add')
            repo = SubElement(add, 'repo')
            repos = SubElement(repo, 'repos')
            for source in sources:
                source_pkg = SubElement(repos, 'source_pkg')
                source_pkg.text = source
            filename_element = SubElement(repo, 'filename')
            filename_element.text = filename
            return tostring(manifest, encoding='utf-8').decode()

        sources = [
            'deb http://example.com/ focal main restricted universe',
            'deb-src http://example.com/ focal-security main'
        ]
        expected = construct_manifest_xml(sources, 'intel-gpu-jammy.list')
```